core/src/apps/monero/signing/step_09_sign_input.py

```diff
@@ -171,13 +187,88 @@ async def sign_input(
         mg_buffer,
     )
 
-    del (input_secret_key, pseudo_out_alpha, mlsag, ring_pubkeys)
-    state.mem_trace(5, True)
+    del (input_secret_key, pseudo_out_alpha, mlsag, ring_pubkeys, kLRki)
+    state.mem_trace(6, True)
 
     from trezor.messages.MoneroTransactionSignInputAck import (
         MoneroTransactionSignInputAck,
     )
 
+    # Encrypt signature, reveal once protocol finishes OK
+    if state.client_version >= 3:
+        utils.unimport_end(mods)
+        state.mem_trace(7, True)
+        mg_buffer = _protect_signature(state, mg_buffer)
+
+    state.mem_trace(8, True)
+    state.last_step = state.STEP_SIGN
     return MoneroTransactionSignInputAck(
         signature=mg_buffer, pseudo_out=crypto.encodepoint(pseudo_out_c)
```

Comment:
We changed protobuf so it accepts list of bytes as bytes as well. We need to do it this way to avoid memory fragmentation problems and to have random access to MLSAG signature as it is being generated.